driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -177,7 +182,12 @@ protected EncodingMetadata encodeMessageBodyWithMetadata(final BsonOutput bsonOu
             commandStartPosition = bsonOutput.getPosition();
 
             if (payload == null) {
```

Comment:
Could the payload be part of a transaction? If so should it be added here?

Reply:
Yeah, this is confusing.  Here's the issue: for the most part passing serverApi to OP_QUERY (i.e. pre 3.6 servers) is irrelevant, because those servers don't support server api (or transactions) anyway.  But we do need to pass server api via OP_QUERY in one circumstance: when sending the initial isMaster command as part of the handshake.  Because we don't know what server version version it is until we get the isMaster response, we always use OP_QUERY for that isMaster command.  For simplicity, and because we don't actually know the command name here until we serialize it, this code just always adds server api for regular OP_QUERY commands.  But doing the same for ones with payload would require more work, and wouldn't give a benefit, because the server we'd be sending it to doesn't support server api anyway.

Maybe an inline comment explaining this would help?

Reply:
Ack - that makes sense. Its worth a comment explaining its added incase its an isMaster call.

Reply:
Done